src/Symfony/Component/Security/Csrf/TokenStorage/SessionTokenStorage.php

```diff
@@ -45,62 +53,95 @@ public function __construct(SessionInterface $session, string $namespace = self:
      */
     public function getToken(string $tokenId)
     {
-        if (!$this->session->isStarted()) {
-            $this->session->start();
+        if (null === $session = $this->getSession()) {
+            throw new \LogicException('You can not use the CSRF if sessions are disabled. Enable them in "config/packages/framework.yaml".');
+        }
+        if (!$session->isStarted()) {
+            $session->start();
         }
 
-        if (!$this->session->has($this->namespace.'/'.$tokenId)) {
+        if (!$session->has($this->namespace.'/'.$tokenId)) {
             throw new TokenNotFoundException('The CSRF token with ID '.$tokenId.' does not exist.');
         }
 
-        return (string) $this->session->get($this->namespace.'/'.$tokenId);
+        return (string) $session->get($this->namespace.'/'.$tokenId);
     }
 
     /**
      * {@inheritdoc}
      */
     public function setToken(string $tokenId, string $token)
     {
-        if (!$this->session->isStarted()) {
-            $this->session->start();
+        if (null === $session = $this->getSession()) {
+            throw new \LogicException('You can not use the CSRF if sessions are disabled. Enable them in "config/packages/framework.yaml".');
+        }
+        if (!$session->isStarted()) {
+            $session->start();
         }
 
-        $this->session->set($this->namespace.'/'.$tokenId, $token);
+        $session->set($this->namespace.'/'.$tokenId, $token);
     }
 
     /**
      * {@inheritdoc}
      */
     public function hasToken(string $tokenId)
     {
-        if (!$this->session->isStarted()) {
-            $this->session->start();
+        if (null === $session = $this->getSession()) {
+            throw new \LogicException('You can not use the CSRF if sessions are disabled. Enable them in "config/packages/framework.yaml".');
+        }
+        if (!$session->isStarted()) {
+            $session->start();
         }
 
-        return $this->session->has($this->namespace.'/'.$tokenId);
+        return $session->has($this->namespace.'/'.$tokenId);
     }
 
     /**
      * {@inheritdoc}
      */
     public function removeToken(string $tokenId)
     {
-        if (!$this->session->isStarted()) {
-            $this->session->start();
+        if (null === $session = $this->getSession()) {
+            throw new \LogicException('You can not use the CSRF if sessions are disabled. Enable them in "config/packages/framework.yaml".');
+        }
```

Comment:
As this code is duplicate whenever `$this->getSession()` is used, what about throwing the exception in `$this->getSession()` directly?

Reply:
great idea. So much simpler, and I can provide more accurate exception message